pkg/clients/aws/client_test.go

```diff
@@ -0,0 +1,94 @@
+
+	"github.com/stretchr/testify/assert"
+)
+
+const fakeAccountNum = "123456789101"
+
+func TestGetRancherLicense(t *testing.T) {
+	tests := []struct {
+		name              string // name of the test, to be displayed on failure
+		hasAmeaLicense    bool   // if the account has a license for the AMEA product sku
+		hasNonAmeaLicense bool   // if the account has a licensed for the non AMEA product sku
+		includeProductSku bool   // if the return from aws should include or exclude a product sku
+		desiredLicense    string // which license our client should pick - amea, nona-amea, or nothing
+		errDesired        bool   // if we wanted an error for this test case
+	}{
+		{
+			name:              "test amea license",
+			hasAmeaLicense:    true,
+			hasNonAmeaLicense: false,
+			includeProductSku: true,
+			desiredLicense:    rancherProductSKUAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test non-amea license",
+			hasAmeaLicense:    false,
+			hasNonAmeaLicense: true,
+			includeProductSku: true,
+			desiredLicense:    rancherProductSKUNonAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test amea + non-amea license - should not occur in reality",
+			hasAmeaLicense:    true,
+			hasNonAmeaLicense: true,
+			includeProductSku: true,
+			desiredLicense:    rancherProductSKUAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test no valid license",
+			hasAmeaLicense:    false,
+			hasNonAmeaLicense: false,
+			includeProductSku: false,
+			desiredLicense:    "",
+			errDesired:        true,
+		},
+		{
+			name:              "test no product sku amea",
+			hasAmeaLicense:    true,
+			hasNonAmeaLicense: false,
+			includeProductSku: false,
+			desiredLicense:    rancherProductSKUAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test no product sku non-amea",
+			hasAmeaLicense:    false,
+			hasNonAmeaLicense: true,
+			includeProductSku: false,
+			desiredLicense:    rancherProductSKUNonAmea,
+			errDesired:        false,
+		},
+	}
+	for _, test := range tests {
+		mockLMClient := mockLicenseManagerClient{}
+		client := &client{
+			acctNum: fakeAccountNum,
+			lm:      &mockLMClient,
+			sts:     &mockSTSClient{accountNumber: fakeAccountNum},
+		}
+		if test.hasAmeaLicense {
+			mockLMClient.AddLicenseForSku(rancherProductSKUAmea, fakeAccountNum, test.includeProductSku)
+		}
+		if test.hasNonAmeaLicense {
+			mockLMClient.AddLicenseForSku(rancherProductSKUNonAmea, fakeAccountNum, test.includeProductSku)
+		}
+
+		license, err := client.GetRancherLicense(context.TODO())
+		if test.errDesired {
+			assert.NotNil(t, err, "expected an error but err was nil for %s", test.name)
+		} else {
+			assert.Nil(t, err, "no error was expected, but got an error for %s", test.name)
+			assert.NotNil(t, license, "expected a valid license but was nil for %s", test.name)
+			assert.Equal(t, *license.ProductSKU, test.desiredLicense, "received unexpected product sku for %s", test.name)
```

Comment:
we do not need to print the test name in the failure message. The `assert` package will already print it out if the test fails or you test with verbose.

Reply:
Unfortunately, I don't think that this is true due to the structure of the test objects. When I run a purposeful failure for this last statement without the name, I get

```
--- FAIL: TestGetRancherLicense (0.00s)
    client_test.go:90: 
                Error Trace:    client_test.go:90
                Error:          Not equal: 
                                expected: "0b87d4fa-d1fe-41d8-830b-67d4ec381549"
                                actual  : "d3ca94f5-fb96-44c1-a094-415fc0463605"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -0b87d4fa-d1fe-41d8-830b-67d4ec381549
                                +d3ca94f5-fb96-44c1-a094-415fc0463605
                Test:           TestGetRancherLicense
                Messages:       received unexpected product sku
FAIL
```

You'll note that there's no indication of the test name. I'm guessing that I could refactor this to work properly, but given the rapid timeline on this issue and the cosmetic nature of such a change, I'm not sure that it's worth the time. LMK if you disagree.

Reply:
Oh, I didn't notice you weren't using [T.Run()](https://pkg.go.dev/testing#T.Run), which splits each test into its own subtest of TestGetRancherLicenese . Since we are using table-driven test cases here, we want to treat each test entry as a sub-test with `T.Run()`.

Reply:
I've adjusted this to use the run functionality that you suggested here.